codec/src/main/java/io/netty/handler/codec/DefaultTextHeaders.java

```diff
@@ -396,4 +401,188 @@ public TextHeaders clear() {
     private static Comparator<CharSequence> comparator(boolean ignoreCase) {
         return ignoreCase ? CHARSEQUENCE_CASE_INSENSITIVE_ORDER : CHARSEQUENCE_CASE_SENSITIVE_ORDER;
     }
+
+    /*
+     * This interface enables different implementations for adding/setting header values.
+     * Concrete implementations can control how values are added, for example to add all
+     * values for a header as a comma separated string instead of adding them as multiple
+     * headers with a single value.
+     */
+    private interface ValuesComposer {
+        TextHeaders add(CharSequence name, CharSequence value);
+        TextHeaders add(CharSequence name, CharSequence... values);
+        TextHeaders add(CharSequence name, Iterable<? extends CharSequence> values);
+
+        TextHeaders addObject(CharSequence name, Object value);
```

Comment:
I don't think we need to override this.  All `add*(...)` methods that don't have collection parameters should be calling `add(...)`.  See [DefaultHeaders.java#L504](https://github.com/netty/netty/blob/master/codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java#L504).